services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -590,6 +595,57 @@ private Set<String> getRegionsToRollForward(
     return completedNonTargetRegions;
   }
 
+  private boolean didPostVersionSwapValidationsPass(
+      Store parentStore,
+      int targetVersionNum,
+      String clusterName,
+      String targetRegion,
+      ReadWriteStoreRepository repository,
+      String kafkaTopicName) {
+    Boolean proceed = true;
+    List<LifecycleHooksRecord> storeLifecycleHooks = parentStore.getStoreLifecycleHooks();
+    for (LifecycleHooksRecord lifecycleHooksRecord: storeLifecycleHooks) {
+      StoreLifecycleHooks storeLifecycleHook = ReflectUtils.callConstructor(
+          ReflectUtils.loadClass(lifecycleHooksRecord.getStoreLifecycleHooksClassName()),
+          new Class<?>[] { VeniceProperties.class },
+          new Object[] { veniceControllerMultiClusterConfig.getCommonConfig().getProps() });
+
+      Properties properties = new Properties();
+      properties.putAll(lifecycleHooksRecord.getStoreLifecycleHooksParams());
+      VeniceProperties veniceProperties = new VeniceProperties(properties);
+      StoreVersionLifecycleEventOutcome outcome = storeLifecycleHook.postStoreVersionSwap(
+          clusterName,
+          parentStore.getName(),
+          targetVersionNum,
+          targetRegion,
+          null,
+          veniceProperties);
+
+      if (StoreVersionLifecycleEventOutcome.WAIT.equals(outcome)) {
+        String message = "Skipping version swap for store: " + parentStore.getName() + " on version: "
+            + targetVersionNum + " as post version swap validations emitted WAIT";
+        logMessageIfNotRedundant(message);
+      } else if (StoreVersionLifecycleEventOutcome.ROLLBACK.equals(outcome)) {
+        String message = "Skipping version swap for store: " + parentStore.getName() + " on version: "
+            + targetVersionNum + "as post version swap validations emitted a roll back";
+        logMessageIfNotRedundant(message);
+
+        veniceParentHelixAdmin.rollbackToBackupVersion(clusterName, parentStore.getName(), targetRegion);
+        parentStore.updateVersionStatus(targetVersionNum, ERROR);
+        repository.updateStore(parentStore);
+
+        LOGGER.info("Truncating kafka topic: {} after validations emitted a roll back", kafkaTopicName);
```

Comment:
can you please add a TODO to delete the following code after parent VT deprecation